src/app/core/services/httperror-interceptor.service.ts

```diff
@@ -1,33 +1,59 @@
 import { Injectable } from '@angular/core';
 import { HttpErrorResponse, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
 import { throwError } from 'rxjs';
-import { catchError } from 'rxjs/operators';
+import { catchError, retryWhen, delay, takeWhile } from 'rxjs/operators';
+import { AppConfigService } from './app-config.service';
 
-@Injectable({providedIn: 'root' })
+
+@Injectable({ providedIn: 'root' })
 export class HttpErrorInterceptorService implements HttpInterceptor {
-    intercept(request: HttpRequest<any>, next: HttpHandler) {
-        return next.handle(request)
-        .pipe(
-            catchError((error: HttpErrorResponse) => {
-                const errorMessage = this.setError(error);
-                return throwError(error.error);
-            })
-        );
-    }
 
-     setError(error: HttpErrorResponse) {
-        let errorMessage = 'Unknown error occured!';
-        if (error.error instanceof ErrorEvent) {
-          // Client-side errors
-          errorMessage = `Error: ${error.error.message}`;
-        } else {
-          // Server-side errors
-          if(error.status!==0){
-            errorMessage = error.error.message;
-          }
-          
-        }
-        return errorMessage;
+  essBaseUrl = AppConfigService.settings['essConfig'].apiUrl;
+  maxRetries = AppConfigService.settings['essConfig'].maxRetries;
+  initialDelayMs = AppConfigService.settings['essConfig'].initialDelayMs;
+  retryCount = 0;
+
+  intercept(request: HttpRequest<any>, next: HttpHandler) {
+    return next.handle(request)
+      .pipe(
+        retryWhen(errors =>
+          errors.pipe(
+            takeWhile((err: HttpErrorResponse, index) => {
+              if (request.url.includes(this.essBaseUrl) &&
+                        (this.retryCount < this.maxRetries - 1)  &&
+                        (err.status != 400 && err.status != 415)) {
```

Comment:
I think there are many more 4xx codes that shouldn't be retried

In fact the API retries only for 5xx and 408 responses

For example we don't want to retry for 403 forbidden and most of the other 400 responses

If possible could you please restructure this code so we retry for 5xx and 408 responses only

Reply:
Continue with exp. back off for 5xx, 408 + 429. Ignore the 'retry after' duration for the 429 case and continue with exp. back off

Reply:
Please delay for 1, 2, 4, 8 seconds

Reply:
suggestions implemented. Thanks